### PR TITLE
Traffic lights v2

### PR DESCRIPTION
**Pull request by strump**
Finishing PR #7281 by @RedAuburn

Rebased his branch to `master`. And fixed `drules_merge.py` script (this also affected speed cameras).

Screenshot 2024-07-29 at 18 01 33

_(This block https://osm.org/?mlat=51.48147&mlon=-0.00903 on the screenshot)_

Map with traffic lights: [UK_England_Greater London.mwm](https://drive.proton.me/urls/R523V2FFWM#rj3ReD7UrBTC)

### CODE REVIEW

File: tools/python/stylesheet/drules_merge.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def create_diff(zooms1, zooms2):
     print("{}: missing completely; {}".format(typ, zooms_string(zooms2[typ][0].scale, zooms2[typ][1].scale)))
     cont = drules_struct_pb2.ClassifElementProto()
     cont.name = typ
-    for z in range(zooms2[typ][0].scale, zooms2[typ][1].scale):
+    for z in range(zooms2[typ][0].scale, zooms2[typ][1].scale+1):
```
Comment (strump) [inline tools/python/stylesheet/drules_merge.py:77]: > There are 2 hard problems in computer science: cache invalidation, naming things, and off-by-1 errors.

File: data/replaced_tags.txt [hunk 2]
```diff
@@ -149,8 +149,6 @@ artwork_type=bust : artwork_type=statue
 
 place=allotments : place=neighbourhood
 
-crossing=traffic_signals : highway=traffic_signals
```
Comment (biodranik) [inline data/replaced_tags.txt:152]: Crossing traffic lights will be ignored now, correct? They should be added separately, right? @euf can we do a red + green icon for crossings?

File: data/styles/vehicle/include/Icons.mapcss [hunk 3]
```diff
@@ -124,6 +124,10 @@ node|z14-[aeroway=aerodrome]::int_name,
 
 /*check offset int_name*/
 
+/* 3.2 Road Icons */
+node|z19-[highway=traffic_signals],
```
Comment (biodranik) [inline data/styles/vehicle/include/Icons.mapcss:128]: For vehicle style, does it make sense to show traffic lights earlier? Will they be visible at all on mobile devices when the camera is up at the city speed of 30-50 mph?

File: tools/python/stylesheet/drules_merge.py (hunk 1 above)
Comment (biodranik) [inline tools/python/stylesheet/drules_merge.py:77]:
```suggestion
    for z in range(zooms2[typ][0].scale, zooms2[typ][1].scale + 1):
```

**Review by biodranik — APPROVED** [review]
@vng PTAL, I'm ok with traffic lights for vehicles only as the first step. The same type will be used to penalize routing too, right?

**Review by vng — APPROVED** [review]

**Comment by RedAuburn** [comment]
thank you! :)

File: data/replaced_tags.txt (hunk 2 above)
Comment (euf) [reply data/replaced_tags.txt:152]: Like a pedestrian oriented traffic lights you mean? Sure we can 😄

**Pull request merged**